Sources/Vapor/HTTP/Server/HTTPServer.swift

```diff
@@ -447,17 +447,47 @@ private final class HTTPServerConnection {
     }
 }
 
+/// A simple channel handler that catches errors emitted by parsing HTTP requests
+/// and sends 400 Bad Request responses.
+///
+/// This channel handler provides the basic behaviour that the majority of simple HTTP
+/// servers want. This handler does not suppress the parser errors: it allows them to
+/// continue to pass through the pipeline so that other handlers (e.g. logging ones) can
+/// deal with the error.
+/// 
+/// adapted from: https://github.com/apple/swift-nio/blob/00341c92770e0a7bebdc5fda783f08765eb3ff56/Sources/NIOHTTP1/HTTPServerProtocolErrorHandler.swift
 final class HTTPServerErrorHandler: ChannelInboundHandler {
     typealias InboundIn = Never
+    typealias OutboundOut = Response
     let logger: Logger
+    private var hasUnterminatedResponse: Bool = false
     
     init(logger: Logger) {
         self.logger = logger
     }
     
     func errorCaught(context: ChannelHandlerContext, error: Error) {
-        self.logger.debug("Unhandled HTTP server error: \(error)")
-        context.close(mode: .output, promise: nil)
+        guard error is HTTPParserError else {
+            context.fireErrorCaught(error)
+            return
+        }
+
+        // Any HTTPParserError is automatically fatal, and we don't actually need (or want) to
+        // provide that error to the client: we just want to tell it that it screwed up and then
+        // let the rest of the pipeline shut the door in its face. However, we can only send an
+        // HTTP error response if another response hasn't started yet.
```

Comment:
I'm not sure we want to use this tone of voice in code comments / docs
```suggestion
        // Any HTTPParserError is automatically fatal, and we don't actually need (or want) to
        // provide that error to the client: we just want to inform them something went wrong and then
        // let the rest of the pipeline close. However, we can only send an
        // HTTP error response if another response hasn't started yet.
```

Reply:
@MrSkwiggs thanks for having a look at the PR.

I actually took this comment directly from https://github.com/apple/swift-nio/blob/00341c92770e0a7bebdc5fda783f08765eb3ff56/Sources/NIOHTTP1/HTTPServerProtocolErrorHandler.swift#L41, but I think it is perfectly fine to adapt to Vapors code/documentation style.

Reply:
Huh interesting 😅

Reply:
While I may be a bit late to the table, NIO is never shy about accepting improvements of this sort, and Vapor isn't either; the suggested rephrasing is a definite improvement.

Reply:
I adjusted the comment as suggested